### PR TITLE
provide git information with --version

### LINKED ISSUE #1848: mitmproxy sysinfo: show git tag/ref/commit
It would be nice if `mitmproxy --sysinfo` would also display the current git tag/ref/commit when running from source. Currently only `0.19` is displayed, which makes it a bit undeterministic what code is actually running. We can probably reuse some code from rtool here.

### PR DESCRIPTION
**Pull request by Kriechi**
fixes #1848

### CODE REVIEW

File: mitmproxy/utils/debug.py [hunk 1]
```diff
@@ -5,18 +5,24 @@
...
 
 from mitmproxy import version
 
 from OpenSSL import SSL
 
 
-def sysinfo():
+def dump_system_info():
+    old_dir = os.getcwd()
+    os.chdir(os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
```
Comment (mhils) [inline mitmproxy/utils/debug.py]: I think we can re-use this: https://github.com/mitmproxy/mitmproxy/blob/490872ebefd87e9a55f2db3dd18c0fbb58dbf206/mitmproxy/test/tutils.py#L22

File: mitmproxy/utils/debug.py [hunk 2]
```diff
@@ -5,18 +5,24 @@
...
+    git_describe = subprocess.check_output(['git', 'describe', '--tags', '--long']).decode().strip()
```
Comment (mhils) [inline mitmproxy/utils/debug.py]: Let's wrap this in a catch-all try-except block. Users may not have git installed, etc. etc. If it fails, we can still display everything else.

File: mitmproxy/utils/debug.py (hunk 1 above)
Comment (Kriechi) [inline mitmproxy/utils/debug.py]: Where would you put that function if we want to move it from `test/` into `mitmproxy/`?

Comment (mhils) [inline mitmproxy/utils/debug.py]: mitmproxy/utils?

Comment (Kriechi) [inline mitmproxy/utils/debug.py]: Done.

**Pull request merged**